dora/core/common/src/main/java/alluxio/uri/UfsUrl.java

```diff
@@ -1,18 +1,87 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
 package alluxio.uri;
 
 import alluxio.AlluxioURI;
 import alluxio.grpc.UfsUrlMessage;
+
+import com.google.common.base.Preconditions;
 import org.apache.logging.log4j.util.Strings;
 
+import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
 
 public class UfsUrl {
+
+  // TODO(Tony Sun): replace each xx_separator by the static variables.
+  public static final String SCHEME_SEPERATOR = "://";
+  public static final String PATH_SEPERATOR = "/";
   UfsUrlMessage mProto;
 
   public UfsUrl(UfsUrlMessage proto) {
+    Preconditions.checkArgument(proto.getPathComponentsList().size() != 0,
+        "the proto.path is empty, please check the proto first");
     mProto = proto;
   }
 
+  public UfsUrl(String ufsPath) {
+    // TODO(Tony Sun): Considering the case below:
+    //  when scheme does not exist, how to determine the scheme, or a empty scheme.
+    Preconditions.checkArgument(!ufsPath.isEmpty(),
+            "ufsPath is empty, please input a non-empty ufsPath.");
+    List<String> preprocessingPathList = Arrays.asList(ufsPath.split(SCHEME_SEPERATOR));
+
+    // TODO(Tony Sun): Does every input path contain authority?
+    String scheme = preprocessingPathList.get(0);
+    String authorityAndPath = null;
+    // If ufsPath has no '://', then the preprocessingPathList has only one elem.
+    // TODO(Tony Sun): Refactor it later. Now default ufs is local, the design is ugly.
+    if (preprocessingPathList.size() == 1)  {
+      scheme = "file";
+      authorityAndPath = preprocessingPathList.get(0);
+    }
+    else if (preprocessingPathList.size() == 2) {
+      authorityAndPath = preprocessingPathList.get(1);
+    }
+    // TODO(Tony Sun): what if preprocessingPathList.size() > 2? Fix it!
+    int indexOfFirstSlashAfterAuthority = authorityAndPath.indexOf(PATH_SEPERATOR);
+    // Empty path is excluded here.
+    Preconditions.checkArgument(indexOfFirstSlashAfterAuthority != -1,
+        "Please input a valid path.");
+    String authorityString = authorityAndPath.substring(0, indexOfFirstSlashAfterAuthority);
+    String pathString = authorityAndPath.substring(indexOfFirstSlashAfterAuthority);
+
+    // TODO(Tony Sun): Is the input of UfsUrl the same as the input of AlluxioUri? Can I use
+    //  the string test format like AlluxioURITest.java?
+
+    // TODO(Tony Sun): Do we need to handle path like '/////path/to/dir'?
```

Comment:
We don't need to handle illegal paths like '/////path/to/dir'

I think we can perform only basic checks in the Alluxio layer. By "basic" I mean cover all corner cases `AlluxioURITest` already raises and covers.

For other strange corner cases, Alluxio passes the path to UFS to access it, and UFS will throw an exception if the path is illegal. Alluxio just needs to properly propagate the exception back to the client, so the client  knows the input param is wrong.